conan/tools/cmake/cmake.py

```diff
@@ -60,7 +60,7 @@ def __init__(self, conanfile):
         self._toolchain_file = configure_preset.get("toolchainFile")
         self._cache_variables = configure_preset["cacheVariables"]
 
-        self._cmake_program = "cmake"  # Path to CMake should be handled by environment
+        self._cmake_program = conanfile.conf.get("tools.cmake:path", check_type=str, default="cmake")
```

Comment:
This feature would be in contradiction with the many existing ``tool_requires`` to ``cmake``, without any validation of the version, which is a bit dangerous.

Probably it is also not aligned with conan 2.0 principle that ``tool_requires`` can affect the ``package_id``.
I think a version-checking approach, with explicit modeling of the ``tool_requires`` and valid for all ``tool_requires`` would be something like: https://github.com/conan-io/conan/pull/10166 (still the desired version has to be in the Path, but I think that is the responsibility of the users).

Let's discuss the different approaches with the team too.